sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +766,210 @@ def limiting_distribution(self):
         eqs.append(sum(wi) - 1)
         soln = list(linsolve(eqs, wi))[0]
         return ImmutableMatrix([[sol for sol in soln]])
+
+
+class BernoulliProcess(DiscreteTimeStochasticProcess):
+    """
+    The Bernoulli process consists of repeated
+    independent Bernoulli trials with the same parameter `p`.
+    It's assumed that the probability `p` applies to every
+    trial and that the outcomes of each trial
+    are independent of all the rest. Therefore Bernoulli Processs
+    is Discrete State and Discrete Time Stochastic Process.
+
+    Parameters
+    ==========
+
+    sym: Symbol/string_types
+    p: Proability of getting success
+
+    Examples
+    ========
+
+    >>> from sympy.stats import BernoulliProcess, P, E
+    >>> B = BernoulliProcess("B", 0.7)
+
+    >>> B.state_space
+    FiniteSet(0, 1)
+
+    >>> (B.p).round(2)
+    0.70
+
+    >>> (B.probability_r_success(ntrials=5, rsuccess=5,evaluate=True)).round(2)
+    0.17
+
+    >>> (B.probability_of_rth_success(ntrial=5, rsuccess=3)).round(2)
```

Comment:
https://github.com/sympy/sympy/pull/18173/files#r362534453 and https://github.com/sympy/sympy/pull/18173/files#r362535646

Reply:
Are other things good?

Reply:
Other part of the PR look good to me. What actually should happen is that probability queries should be made through `P` as much as possible. So, wherever possible please make a change.

Reply:
Yes surely. Thanks!